src/main/java/de/tum/in/www1/artemis/service/DataExportService.java

```diff
@@ -0,0 +1,461 @@
+package de.tum.in.www1.artemis.service;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.stream.Stream;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.modeling.ModelingSubmission;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseStudentParticipation;
+import de.tum.in.www1.artemis.domain.participation.StudentParticipation;
+import de.tum.in.www1.artemis.domain.quiz.*;
+import de.tum.in.www1.artemis.repository.*;
+import de.tum.in.www1.artemis.service.programming.ProgrammingExerciseExportService;
+import de.tum.in.www1.artemis.web.rest.dto.RepositoryExportOptionsDTO;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+
+/**
+ * Service Implementation for managing the data export in accordance with Art. 15 GDPR.
+ */
+@Service
+public class DataExportService {
+
+    private static final String ZIP_FILE_EXTENSION = ".zip";
+
+    private static final String CSV_FILE_EXTENSION = ".csv";
```

Comment:
Apparently not used?

Reply:
now they are used, I replaced the hardcoded strings with them.